resolver-dns/src/main/java/io/netty/resolver/dns/DnsResolveContext.java

```diff
@@ -792,12 +795,41 @@ private void onExpectedResponse(
                 } while (resolved != null);
 
                 if (resolved == null) {
-                    continue;
+                    assert questionName.isEmpty() || questionName.charAt(questionName.length() - 1) == '.';
+
+                    for (String searchDomain : parent.searchDomains()) {
+                        if (searchDomain.isEmpty()) {
+                            continue;
+                        }
+
+                        final String fqdn;
+                        if (searchDomain.charAt(searchDomain.length() - 1) == '.') {
+                            fqdn = questionName + searchDomain;
```

Comment:
Is it possible that `searchDomain` is equal to `.` and doesn't have any other characters?

Reply:
Yeah could be and I do not see any issues here if that is the case. This will be the same behavior as we appending such a search domain to the query when we do.

Reply:
If the question name was `netty.io.`, `fqdn` will be `netty.io..`

Reply:
Sure, I am perhaps not getting your concern. Can you explain.

Reply:
Example:
questionName: `netty.io.` (with final dot)
searchDomain: `.` (just a single dot)
In this case, we probably should not concatenate those two into `fqdn`, because the result will be illegal (two dots at the end). We should skip and try the next `searchDomain`.

Reply:
> We should skip and try the next searchDomain.

The result is just used to match the name in question name. If it does not match (which presumably it won't in this case), it will effectively be skipped and check the next search domain.

Reply:
ok, sounds good!